app/src/main/java/com/example/wrk/ScratchCreateActivity.java

```diff
@@ -0,0 +1,125 @@
+    EditText etCreateWorkoutTitle;
+    ImageButton ibDeleteWorkout;
+    ImageButton ibAddExercise;
+    Button btnSave;
+    ArrayList<WorkoutComponent> mComponents;
+    RecyclerView rvComponents;
+    ScratchCreateAdapter adapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_scratch_create);
+
+        etCreateWorkoutTitle = findViewById(R.id.etCreateWorkoutTitle);
+        ibDeleteWorkout = findViewById(R.id.ibDeleteWorkout);
+        ibAddExercise = findViewById(R.id.ibAddExercise);
+        btnSave = findViewById(R.id.btnSave);
+        rvComponents = findViewById(R.id.rvCreateComponents);
+
+        mComponents = new ArrayList<>();
+        adapter = new ScratchCreateAdapter(this, mComponents);
+        rvComponents.setAdapter(adapter);
+        rvComponents.setLayoutManager(new LinearLayoutManager(this));
+
+        ibAddExercise.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent i = new Intent(ScratchCreateActivity.this, ExerciseListActivity.class);
+                startActivity(i);
+            }
+        });
+
+        ibDeleteWorkout.setOnClickListener(new View.OnClickListener() {
+            // Create an alert dialog to confirm deletion
+            @Override
+            public void onClick(View v) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(ScratchCreateActivity.this);
+                AlertDialog dialog;
+                builder.setTitle("Delete Template");
+                builder.setMessage("Are you sure you want to delete this template?");
+                builder.setPositiveButton("DELETE", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        finish();
+                    }
+                });
+                builder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        dialog.cancel();
+                    }
+                });
+                dialog = builder.create();
+                dialog.setOnShowListener(new DialogInterface.OnShowListener() {
+                    @Override
+                    public void onShow(DialogInterface dial) {
+                        dialog.getButton(AlertDialog.BUTTON_POSITIVE).setTextColor(Color.RED);
+                    }
+                });
+                dialog.show();
+            }
+        });
+
+        btnSave.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String title = etCreateWorkoutTitle.getText().toString();
+                if (!title.isEmpty()) {
```

Comment:
It's always safer to do null check first before checking the length, to avoid unexpected NullPointerException.
`if(!title != null && !title.isEmpty())`